src/main/java/seedu/address/logic/parser/FindAllTagCommandParser.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.logic.commands.FindAllTagCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.TagsContainAllTagsPredicate;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new FindAllTagCommand object
+ */
+public class FindAllTagCommandParser implements Parser<FindAllTagCommand> {
+    public static final String MESSAGE_INVALID_TAG = "Tags names should be alphanumeric.";
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindAllTagCommand
+     * and returns a FindAllTagCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindAllTagCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindAllTagCommand.MESSAGE_USAGE));
+        }
+
+        String[] tagKeywords = trimmedArgs.split("\\s+");
```

Comment:
How are we splitting the tags? Do we allow tags that contain multiple words?

Reply:
Previously the tags are alpha-numeric, so there should not be any space or dash in between word. We are still discussing if we allow tags to have space or dash. A tag should not be so long, so I'm not sure if we need to have space; if we allow dash, it may make users confuse between the command and tags